src/window/building/depot.c

```diff
@@ -0,0 +1,441 @@
+
+#include "building/building.h"
+#include "building/storage.h"
+#include "building/warehouse.h"
+#include "figure/figure.h"
+#include "graphics/generic_button.h"
+#include "graphics/image.h"
+#include "graphics/lang_text.h"
+#include "graphics/panel.h"
+#include "graphics/screen.h"
+#include "graphics/scrollbar.h"
+#include "graphics/text.h"
+#include "graphics/window.h"
+#include "translation/translation.h"
+#include "window/building_info.h"
+
+static void order_set_source(int index, int param2);
+static void order_set_destination(int index, int param2);
+static void order_set_resource(int index, int param2);
+static void order_set_condition_type(int index, int param2);
+static void order_set_condition_threshold(int index, int param2);
+static void set_order_resource(int depot_building_id, int resource_id);
+
+static void setup_for_selected_depot(building_info_context *c);
+
+#define DEPOT_BUTTONS_X_OFFSET 32
+#define DEPOT_BUTTONS_Y_OFFSET 204
+#define ROW_HEIGHT 22
+#define MAX_VISIBLE_ROWS 15
+
+static struct {
+    int focus_button_id;
+    int orders_focus_button_id;
+    int resource_focus_button_id;
+    int storage_building_focus_button_id;
+    int depot_resource_focus_button_id;
+    int depot_building_id;
+    resource_type target_resource_id;
+} data;
+
+static const translation_key TRANSLATION_KEY_ORDER_CONDITION[4] = {
+    TR_ORDER_CONDITION_NEVER,
+    TR_ORDER_CONDITION_ALWAYS,
+    TR_ORDER_CONDITION_SOURCE_HAS_MORE_THAN,
+    TR_ORDER_CONDITION_DESTINATION_HAS_LESS_THAN
+};
+
+static void on_scroll(void);
+static scrollbar_type scrollbar = { 0, 0, ROW_HEIGHT * MAX_VISIBLE_ROWS, 432, MAX_VISIBLE_ROWS, on_scroll, 0, 4 };
+
+// field values will be overwritten in draw_depot_select_source_destination
+static generic_button depot_select_storage_buttons[] = {
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+};
+
+static generic_button depot_select_resource_buttons[] = {
+    {18, 0, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN},
+    {232, 0, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 1},
+    {18, 26, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 2},
```

Comment:
Why not also set a `ROW_HEIGHT_RESOURCE` to `26` and prevent using all the multiples of 26 here?

Reply:
Done.